lms/djangoapps/mobile_api/models.py

```diff
@@ -25,3 +25,57 @@ def get_video_profiles(cls):
         Get the list of profiles in priority order when requesting from VAL
         """
         return [profile.strip() for profile in cls.current().video_profiles.split(",") if profile]
+
+
+class AppVersionConfig(ConfigurationModel):
+    """
+    Configuration for mobile app versions available.
+    """
+    PLATFORM_CHOICES = tuple([
+        (platform, platform)
+        for platform in PLATFORM_CLASSES.keys()
+    ])
+    KEY_FIELDS = ('platform', 'version')  # combination of mobile platform and version is unique
+    platform = CharField(max_length=50, choices=PLATFORM_CHOICES, blank=False)
+    version = CharField(
+        max_length=50,
+        blank=False,
+        help_text="Version should be in the format X.X.X.Y where X is a number and Y is alphanumeric"
+    )
+    major_version = IntegerField()
+    minor_version = IntegerField()
+    patch_version = IntegerField()
+    expire_at = DateTimeField(null=True, blank=True, verbose_name="Expiry date for platform version")
+
+    class Meta:
+        ordering = ['-major_version', '-minor_version', '-patch_version']
+
+    def __unicode__(self):
+        return "{}_{}".format(self.platform, self.version)
+
+    @classmethod
+    def latest_version(cls, platform):
+        """ returns latest supported version for a platform """
+        latest_version_config = cls.objects.current_set().filter(platform=platform, enabled=True).first()
+        if latest_version_config:
+            return latest_version_config.version
+
+    @classmethod
+    def last_supported_date(cls, platform, version):
+        """ returns date when version will get expired for a platform """
+        parsed_version = utils.parsed_version(version)
+        active_configs = cls.objects.current_set().filter(
+            platform=platform,
+            enabled=True,
+            expire_at__isnull=False
+        ).reverse()
```

Comment:
Don't you want to sort this result set by version?  Otherwise, you'll get an unexpected result when you loop through the versions in the subsequent lines.